controllers/manager/gatewayvmconfiguration_controller.go

```diff
@@ -85,9 +121,48 @@ func (r *GatewayVMConfigurationReconciler) Reconcile(ctx context.Context, req ct
 func (r *GatewayVMConfigurationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&egressgatewayv1alpha1.GatewayVMConfiguration{}).
+		Watches(&corev1.Node{}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(resourceHasFilterLabel(consts.AKSNodepoolModeLabel, consts.AKSNodepoolModeValue))).
```

Comment:
Upstream usage may not have these AKS managed labels, let's filter label `"kubeegressgateway.azure.com/mode": "true"` also?